client/lib/post-normalizer/rule-pick-canonical-image.js

```diff
@@ -6,40 +6,30 @@ import { find } from 'lodash';
 /**
  * Internal Dependencies
  */
-import { thumbIsLikelyImage, isCandidateForCanonicalImage } from './utils';
-
-
+import { imageSizeFromAttachments, thumbIsLikelyImage, isCandidateForCanonicalImage } from './utils';
 
 export default function pickCanonicalImage( post ) {
-	let canonicalImage;
-	if ( post.canonical_image ) {
-		post.canonical_image = null;
-	}
-	if ( post.images ) {
-		canonicalImage = find( post.images, isCandidateForCanonicalImage );
+	if ( thumbIsLikelyImage( post.post_thumbnail ) ) {
+		const { URL: url, width, height } = post.post_thumbnail;
+		post.canonical_image = {
+			uri: url,
+			width,
+			height
+		};
+	} else if ( post.featured_image ) {
```

Comment:
I'm pretty sure this clause can be removed since the existence of `post.post_thumbnail` coincides 1:1 with the existence of a `featured_image`